creator-node/test/ipfsClient.test.js

```diff
@@ -0,0 +1,228 @@
+const assert = require('assert')
```

Comment:
I think we should actually have some hardcoded files that we compare the CID values to, just to make sure the absolute cid values don't change. here's what i came up with

```js
const assert = require('assert')
const fs = require('fs')
const path = require('path')
const sinon = require('sinon')

const ipfsClient = require('../src/ipfsClient')
const ipfsAdd = require('../src/ipfsAdd')

const ipfs = ipfsClient.ipfs
const ipfsLatest = ipfsClient.ipfsLatest

describe('test ipfsClient', async () => {
  afterEach(() => {
    sinon.restore()
  })

  const files = [
    { filePath: path.join(__dirname, './assets/random.png'), cid: 'QmNinpoNbbZAv2g7bsJ9fUeZzWQvt2A8zLpnNsHUSAB3Aj', ipfsOldReadStreamCid: 'QmUNLLsPACCz1vLxQVkXqqLX5R1X345qqfHbsf67hvA3Nn' },
    { filePath: path.join(__dirname, './assets/random.txt'), cid: 'QmRJeoc3PWHjawiSLY43xe4kFw9X7yDySqdxjJrbuMq89R', ipfsOldReadStreamCid: 'QmUNLLsPACCz1vLxQVkXqqLX5R1X345qqfHbsf67hvA3Nn' }
    // ...maybe a few more types
  ]
  // TEST IPFS ADD
  for (let i = 0; i < files.length; i++) {
    const fileData = files[i]

    console.log('fileData', fileData)
    it('[buffer] when adding a buffer, the cids should be consistent', async () => {
      const buffer = fs.readFileSync(fileData.filePath)

      // Original IPFS instance
      const ipfsAddBufferResp = (await ipfs.add(buffer))[0].hash

      // Original IPFS instance with `onlyHash` flag
      const ipfsAddBufferRespOnlyHashFlag = (await ipfs.add(buffer, { onlyHash: true }))[0].hash

      // Newer IPFS instance
      let ipfsLatestAddBufferResp
      for await (const resp of ipfsLatest.add(buffer)) {
        ipfsLatestAddBufferResp = `${resp.cid}`
      }

      // Newer IPFS instance with `onlyHash` flag
      let ipfsLatestAddBufferRespWithOnlyHashFlag
      for await (const resp of ipfsLatest.add(buffer, { onlyHash: true })) {
        ipfsLatestAddBufferRespWithOnlyHashFlag = `${resp.cid}`
      }

      // Only hash logic
      const onlyHash = await ipfsAdd.ipfsOnlyHashNonImages(buffer)

      console.log('buffers', {
        ipfsAddBufferResp,
        ipfsAddBufferRespOnlyHashFlag,
        ipfsLatestAddBufferResp,
        ipfsLatestAddBufferRespWithOnlyHashFlag,
        onlyHash
      })

      const allResults = [
        ipfsAddBufferResp,
        ipfsAddBufferRespOnlyHashFlag,
        ipfsLatestAddBufferResp,
        ipfsLatestAddBufferRespWithOnlyHashFlag,
        onlyHash
      ]

      assert.ok(!!allResults.reduce(function (a, b) { return (a === b) ? a : NaN }))
      allResults.forEach(function (result) { assert.deepStrictEqual(result, fileData.cid) })
    })

    it('[readstream] when adding a readstream, the cids should be consistent', async () => {
      // Original IPFS instance
      console.log('about to readFileStream', fileData.filePath)
      const ipfsAddReadstreamResp = (await ipfs.add(fs.createReadStream(fileData.filePath)))[0].hash

      // Original IPFS instance with `onlyHash` flag
      const ipfsAddReadstreamRespOnlyHashFlag = (await ipfs.add(fs.createReadStream(fileData.filePath), { onlyHash: true }))[0].hash

      // Newer IPFS instance
      let ipfsLatestAddReadstreamResp
      for await (const resp of ipfsLatest.add(fs.createReadStream(fileData.filePath))) {
        ipfsLatestAddReadstreamResp = `${resp.cid}`
      }

      // Newer IPFS instance with `onlyHash` flag
      let ipfsLatestAddReadstreamRespWithOnlyHashFlag
      for await (const resp of ipfsLatest.add(fs.createReadStream(fileData.filePath), { onlyHash: true })) {
        ipfsLatestAddReadstreamRespWithOnlyHashFlag = `${resp.cid}`
      }

      // Only hash logic
      const onlyHash = await ipfsAdd.ipfsOnlyHashNonImages(fs.createReadStream(fileData.filePath))

      console.log('readstream', {
        ipfsAddReadstreamResp,
        ipfsAddReadstreamRespOnlyHashFlag,
        ipfsLatestAddReadstreamResp,
        ipfsLatestAddReadstreamRespWithOnlyHashFlag,
        onlyHash
      })

      // NOTE: we have observed that passing in a readstream to ipfs.add is inconsistent.
      // Commenting out ipfs.add responses in test result comparisons
      const allResults = [
        ipfsLatestAddReadstreamResp,
        ipfsLatestAddReadstreamRespWithOnlyHashFlag,
        onlyHash
      ]

      // Check that every response in `allResults` is equal to each other
      assert.ok(!!allResults.reduce(function (a, b) { return (a === b) ? a : NaN }))
      // And all results are equal to expected CID
      allResults.forEach(function (result) { assert.deepStrictEqual(result, fileData.cid) })

      // NOTE: we have observed that passing in a readstream to ipfs.add is inconsistent.
      // Commenting out ipfs.add responses in test result comparisons
      const allResultsLegacy = [
        ipfsAddReadstreamResp,
        ipfsAddReadstreamRespOnlyHashFlag
      ]

      // Check that every response in `allResultsLegacy` is equal to each other
      assert.ok(!!allResultsLegacy.reduce(function (a, b) { return (a === b) ? a : NaN }))
      // And all results are equal to expected CID, in this case it's an incorrect CID
      allResultsLegacy.forEach(function (result) { assert.deepStrictEqual(result, fileData.ipfsOldReadStreamCid) })
    })

    // Note: ipfsLatest does not have a `addFromFs()` fn
    it('[file path] when adding a file path, the cids should be consistent', async () => {
      // Original IPFS instance
      const ipfsAddFromFsFilePathResp = (await ipfs.addFromFs(fileData.filePath))[0].hash

      // Original IPFS instance with `onlyHash` flag
      const ipfsAddFromFsFilePathRespOnlyHashFlag = (await ipfs.addFromFs(fileData.filePath, { onlyHash: true }))[0].hash

      // Only hash logic
      const buffer = await fs.readFileSync(fileData.filePath)
      const onlyHash = await ipfsAdd.ipfsOnlyHashNonImages(buffer)

      console.log('file path', {
        ipfsAddFromFsFilePathResp,
        ipfsAddFromFsFilePathRespOnlyHashFlag,
        onlyHash
      })

      const allResults = [
        ipfsAddFromFsFilePathResp,
        ipfsAddFromFsFilePathRespOnlyHashFlag,
        onlyHash
      ]

      assert.ok(!!allResults.reduce(function (a, b) { return (a === b) ? a : NaN }))
      allResults.forEach(function (result) { assert.deepStrictEqual(result, fileData.cid) })
    })

    it('[ipfsAddNonImages] passing in a Buffer and `enableIPFSAdd` = true should work', async () => {
      const buffer = Buffer.from(fs.readFileSync(fileData.filePath))

      const onlyHash = await ipfsAdd.ipfsAddNonImages(buffer)
      const ipfsLatestAddWithDaemonResp = await ipfsAdd.ipfsAddNonImages(buffer, {}, {}, true)

      console.log('buffer', {
        onlyHash,
        ipfsLatestAddWithDaemonResp
      })

      assert.deepStrictEqual(onlyHash, ipfsLatestAddWithDaemonResp)
      assert.deepStrictEqual(onlyHash, fileData.cid)
    })

    it('[ipfsAddNonImages] passing in a ReadStream and `enableIPFSAdd` = true should work', async () => {
      const readStream1 = fs.createReadStream(fileData.filePath)
      const onlyHash = await ipfsAdd.ipfsAddNonImages(readStream1)

      const readStream2 = fs.createReadStream(fileData.filePath)
      const ipfsLatestAddWithDaemonResp = await ipfsAdd.ipfsAddNonImages(readStream2, {}, {}, true)

      console.log('readstream', {
        onlyHash,
        ipfsLatestAddWithDaemonResp
      })

      assert.deepStrictEqual(onlyHash, ipfsLatestAddWithDaemonResp)
      assert.deepStrictEqual(onlyHash, fileData.cid)
    })

    it('[ipfsAddNonImages] passing in a source path and `enableIPFSAdd` = true should work', async () => {
      const onlyHash = await ipfsAdd.ipfsAddNonImages(fileData.filePath)
      const ipfsLatestAddWithDaemonResp = await ipfsAdd.ipfsAddNonImages(fileData.filePath, {}, {}, true)

      console.log('source path', {
        onlyHash,
        ipfsLatestAddWithDaemonResp
      })

      assert.deepStrictEqual(onlyHash, ipfsLatestAddWithDaemonResp)
      assert.deepStrictEqual(onlyHash, fileData.cid)
    })

    it('[ipfsAddNonImages] passing in improper content throws', async () => {
      try {
        await ipfsAdd.ipfsAddNonImages('vicky was here')
        throw new Error('passing in improper data should have failed')
      } catch (e) {
        assert.ok(e.toString().includes('Could not convert content into buffer'))
      }
    })

    it('[ipfsAddNonImages] if `ipfsOnlyHashImages()` errors, `ipfsAddNonImages()` throws', async () => {
      sinon.stub(ipfsAdd, 'ipfsOnlyHashNonImages').throws(new Error('failed to generate only hash'))

      try {
        await ipfsAdd.ipfsAddNonImages(fileData.filePath)
        throw new Error('`ipfsAddNonImages` should throw if `ipfsOnlyHashNonImages` fails')
      } catch (e) {
        assert.ok(e.toString().includes('failed to generate only hash'))
      }
    })

    it('[ipfsAddNonImages] if `ipfsOnlyHashImages()` and ipfs daemon hash resp diverges, `ipfsAddNonImages()` throws', async () => {
      sinon.stub(ipfsAdd, 'ipfsOnlyHashNonImages').returns('QmVHxRocoWgUChLEvfEyDuuD6qJ4PhdDL2dTLcpUy3dSC2'
      )

      try {
        await ipfsAdd.ipfsAddNonImages(fileData.filePath, {}, {}, true)
        throw new Error('`ipfsAddNonImages` should throw if only hash and daemon hash diverges')
      } catch (e) {
        assert.ok(e.toString().includes('are not consistent'))
      }
    })
  }
})

```

Reply:
added static unit tests!